Add pmem configuration parameter validation

michalbiesek opened the pull request:
Should be merged after #120 and #121 

In case of redis wasn't compiled with memkindpmem-str-mode can not be set to value different than none

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/redis/122)
<!-- Reviewable:end -->

KFilipek reviewed (commented):
> Reviewed 9 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/redis/122)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jschmieg and @kasiawasiuta)



<!-- Sent from Reviewable.io -->

Commit:
Validation of pmem-str-mode

In case of redis wasn't compiled with memkind allocator
pmem-str-mode can not be set to value different than "none"

KFilipek reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/pmem/redis/122#-:-M-A8Pfu4nRfXegztixu:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kasiawasiuta)



<!-- Sent from Reviewable.io -->
